src/pages/dogs/ReportDogPage.tsx

```diff
@@ -225,116 +190,47 @@ export const ReportDogPage = withAuthenticationRequired(
-                onChange={inputs.contactName.onTextChange}
-                error={!inputs.contactName.isTextValid}
-              />
-              <RTLTextField
-                rows={2}
-                label={AppTexts.reportPage.extraDetails.contactPhone}
-                fullWidth
-                multiline
-                type="text"
-                margin={"normal"}
-                value={inputs.contactPhone.value}
-                onChange={inputs.contactPhone.onPhoneChange}
-                error={!inputs.contactPhone.isPhoneValid}
-              />
-              <RTLTextField
-                rows={2}
-                label={AppTexts.reportPage.extraDetails.contactEmail}
-                fullWidth
-                multiline
-                type="text"
-                margin={"normal"}
-                value={inputs.contactEmail.value}
-                onChange={inputs.contactEmail.onEmailChange}
-                error={!inputs.contactEmail.isEmailValid}
-              />
-              <RTLTextField
-                rows={2}
-                label={AppTexts.reportPage.extraDetails.contactAddress}
-                fullWidth
-                multiline
-                type="text"
-                margin={"normal"}
-                value={inputs.contactAddress.value}
-                onChange={inputs.contactAddress.onTextChange}
-                error={!inputs.contactAddress.isTextValid}
-              />
-              <RTLTextField
-                rows={5}
-                label={AppTexts.reportPage.extraDetails.extraDetails}
-                fullWidth
-                multiline
-                type="text"
-                margin={"normal"}
-                value={inputs.extraDetails.value}
-                onChange={inputs.extraDetails.onTextChange}
-                error={!inputs.extraDetails.isTextValid}
-              />
+              {getInputsData(dogType).map((inputField) => {
+                const selectedInput = inputs[inputField.name];
+                return (
+                  <>
+                    <RTLTextField
+                      key={inputField.name}
+                      label={inputField.label}
+                      type="text"
+                      margin="normal"
+                      fullWidth
+                      multiline={inputField.multiline}
+                      rows={inputField.rows}
+                      // @ts-expect-error
+                      value={selectedInput?.value}
+                      onChange={
+                        // @ts-expect-error
+                        inputField.onChange ?? selectedInput?.onTextChange
+                      }
+                      // @ts-expect-error
+                      error={inputField.error ?? !selectedInput?.isTextValid}
+                    />
+
+                    {/* We need to render the "dogSex" input RIGHT AFTER "dogSize", and "date" input after "location" */}
+                    {inputField.name === "dogSize" && (
```

Comment:
@YanivWein24 I know that you mean well by implementing this, but I think that it adds unnecessary complexity at this stage. We need to be able to make changes here easily, especially since this is [supposed to be a wizard](https://www.figma.com/file/BOc8Y50O4ycUcaDH5Thzzo/Dog-Finder?type=design&node-id=3-23521&mode=design&t=w1y23TljUjZr7R5w-0) - Let's move that back and refactor once we do the wizard. WDYT?